platform-sdk/swirlds-common/src/main/java/com/swirlds/common/wiring/schedulers/DirectTaskScheduler.java

```diff
@@ -0,0 +1,137 @@
+    private final ObjectCounter onRamp;
+    private final ObjectCounter offRamp;
+    private final FractionalTimer busyTimer;
+
+    /**
+     * Constructor.
+     *
+     * @param model                    the wiring model containing this task scheduler
+     * @param name                     the name of the task scheduler
+     * @param uncaughtExceptionHandler the uncaught exception handler
+     * @param onRamp                   an object counter that is incremented when data is added to the task scheduler
+     * @param offRamp                  an object counter that is decremented when data is removed from the task
+     * @param busyTimer                a timer that tracks the amount of time the task scheduler is busy
+     * @param stateless                true if the work scheduled by this object is stateless
+     */
+    public DirectTaskScheduler(
+            @NonNull final WiringModel model,
+            @NonNull final String name,
+            @NonNull final UncaughtExceptionHandler uncaughtExceptionHandler,
+            @NonNull final ObjectCounter onRamp,
+            @NonNull final ObjectCounter offRamp,
+            @NonNull final FractionalTimer busyTimer,
+            final boolean stateless) {
+        super(model, name, stateless ? TaskSchedulerType.DIRECT_STATELESS : TaskSchedulerType.DIRECT, false, true);
+
+        this.uncaughtExceptionHandler = Objects.requireNonNull(uncaughtExceptionHandler);
+        this.onRamp = Objects.requireNonNull(onRamp);
+        this.offRamp = Objects.requireNonNull(offRamp);
+        this.busyTimer = Objects.requireNonNull(busyTimer);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public long getUnprocessedTaskCount() {
+        return -1;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void flush() {
+        throw new UnsupportedOperationException("Direct task schedulers do not support flushing");
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    protected void put(@NonNull final Consumer<Object> handler, @Nullable final Object data) {
+        onRamp.onRamp();
+        busyTimer.activate();
+        try {
+            handler.accept(data);
+        } catch (final Throwable t) {
+            uncaughtExceptionHandler.uncaughtException(Thread.currentThread(), t);
+        }
+        busyTimer.deactivate();
+        offRamp.offRamp();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    protected boolean offer(@NonNull final Consumer<Object> handler, @Nullable final Object data) {
+        final boolean accepted = onRamp.attemptOnRamp();
+        if (!accepted) {
+            return false;
+        }
+
+        busyTimer.activate();
+        try {
+            handler.accept(data);
+        } catch (final Throwable t) {
+            uncaughtExceptionHandler.uncaughtException(Thread.currentThread(), t);
+        }
+        busyTimer.deactivate();
```

Comment:
these could be extracted to a single method to avoid duplication and potential errors

Reply:
done